### PR TITLE
Add dbt metrics submodule

### LINKED ISSUE #26: Add dbt metrics package support
Add [dbt metrics](https://hub.getdbt.com/dbt-labs/metrics/latest/) package support

### PR DESCRIPTION
**Pull request by damian3031**
resolves https://github.com/starburstdata/dbt-trino-utils/issues/26
Add dbt metrics in dbt-trino-utils:
- add dbt-metrics submodule
- bump Trino and SEP versions
- add integration tests. Lots of models from dbt-metrics needs to be overridden with models pointing to `trino__fact_orders_source.csv`, which have date format that is understood by Trino
- add trino shim for dbt-metrics gen_calendar_join macro

### CODE REVIEW

File: integration_tests/dbt_metrics/models/trino__custom_calendar.sql [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+select * from filtered
+
+
+),
+
+final as (
+    select
+        cast(date_day as date) as date_day,
+        {% if target.type == 'bigquery' %}
+            --BQ starts its weeks on Sunday. I don't actually care which day it runs on for auto testing purposes, just want it to be consistent with the other seeds
```
Comment (hovaesco) [inline integration_tests/dbt_metrics/models/trino__custom_calendar.sql]: I guess we can drop it

File: integration_tests/dbt_metrics/dbt_project.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        +enabled: false
+
+      # Overridden by trino__develop_metric
+      develop_metric:
+        +enabled: false
+      # Overridden by trino__simple_develop_metric
+      simple_develop_metric:
+        +enabled: false
+
+      # TODO: Fix and enable
```
Comment (hovaesco) [inline integration_tests/dbt_metrics/dbt_project.yml:58]: More context in TODO or follow up issue

File: integration_tests/dbt_metrics/dbt_project.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  dbt_metrics_integration_tests:
+
+    # Overridden by trino__custom_calendar
+    custom_calendar:
+      +enabled: false
+
+    metric_testing_models:
+      +materialized: table
+
+      # no median function in Trino
```
Comment (hovaesco) [inline integration_tests/dbt_metrics/dbt_project.yml:39]: Perhaps we could use other function to calculate `median` WDYT?

**Review by hovaesco — COMMENTED** [review]
LGTM % some comments

File: integration_tests/dbt_metrics/models/trino__custom_calendar.sql (hunk 1 above)
Comment (damian3031) [reply integration_tests/dbt_metrics/models/trino__custom_calendar.sql]: yes, removed that part

File: integration_tests/dbt_metrics/dbt_project.yml (hunk 2 above)
Comment (damian3031) [reply integration_tests/dbt_metrics/dbt_project.yml:58]: I'll create follow up issue for fixing those ones.

File: integration_tests/dbt_metrics/dbt_project.yml (hunk 3 above)
Comment (damian3031) [reply integration_tests/dbt_metrics/dbt_project.yml:39]: We could use [approx_percentile](https://trino.io/docs/current/functions/aggregate.html#approx_percentile) function. But in order to calculate median for odd and even number of values in the column, we need to add workaround, as using just `approx_percentile(col_name, 0.50)` for even number of values in the column is not working ([workaround source](https://groups.google.com/g/presto-users/c/wXR-oikVUwo/m/3x8R1T3HKgAJ)):

_Create sample table with even nr of rows:_
```sql
CREATE TABLE memory.default.test_table AS (
select 1 AS col_name
union all
select 2 AS col_name
);
```
_Calculate median:_
```sql
SELECT
IF (
    COUNT(*) % 2 = 0,
    (approx_percentile(col_name, 0.499999) + approx_percentile(col_name, 0.50))/2.0,
    approx_percentile(col_name, 0.50)
) AS median_value
FROM memory.default.test_table;
```

So, it is a bit of code to calculate median in Trino. For sure the best way would be to implement `MEDIAN` in Trino. There is issue for that, but it is very stale and gained no attention https://github.com/trinodb/trino/issues/6309
WDYT?

Comment (hovaesco) [reply integration_tests/dbt_metrics/dbt_project.yml:39]: Just create a follow-up issue and include this code.

**Comment by hovaesco** [comment]
Tests are red.

**Review by hovaesco — APPROVED** [review]

File: integration_tests/dbt_metrics/dbt_project.yml (hunk 3 above)
Comment (SeungHuLee) [reply integration_tests/dbt_metrics/dbt_project.yml:39]: Anything wrong with getting median by value_at_quantile(tdigest_agg(col_name), 0.5)

Comment (hovaesco) [reply integration_tests/dbt_metrics/dbt_project.yml:39]: cc @damian3031

Comment (damian3031) [reply integration_tests/dbt_metrics/dbt_project.yml:39]: @SeungHuLee your solution has the same downside as using [approx_percentile](https://trino.io/docs/current/functions/aggregate.html#approx_percentile) alone - it doesn't work if number of rows is even, and two middle values are not the same numbers. In this case, function should return arithmetic mean of these middle values. In your solution, it returns the bigger value.

So, running this query on table which I defined in my previous comment:
```sql
SELECT
value_at_quantile(tdigest_agg(col_name), 0.5) as median_value
FROM memory.default.test_table;
```
returns `2.0`, but correct result is `1.5`.

That's the reason why above mentioned workaround is needed.

Comment (SeungHuLee) [reply integration_tests/dbt_metrics/dbt_project.yml:39]: @damian3031  Thanks for detailed answer! I'm not really a statistics expert, so I was just using tdigest methods according to following:
https://github.com/trinodb/trino/pull/5158
https://github.com/trinodb/trino/issues/4975
https://arxiv.org/pdf/1902.04023.pdf
By the way, when I use the workaround suggested above, for me the median result changes slightly everytime I execute the same query.
Is there more stable or accurate solution suggested?

**Pull request merged**